google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITDatabaseTest.java

```diff
@@ -151,4 +155,26 @@ public void instanceNotFound() {
       assertThat(e.getResourceName()).isEqualTo(nonExistingInstanceId.getName());
     }
   }
+
+  @Test
+  public void numericPrimaryKey()
+      throws InterruptedException, ExecutionException, TimeoutException {
+    final String instanceId = env.getTestHelper().getInstanceId().getInstance();
+    final String databaseId = env.getTestHelper().getUniqueDatabaseId();
+    final DatabaseAdminClient databaseAdminClient =
+        env.getTestHelper().getClient().getDatabaseAdminClient();
+
+    try {
+      final OperationFuture<Database, CreateDatabaseMetadata> op =
+          databaseAdminClient.createDatabase(
+              instanceId,
+              databaseId,
+              Collections.singletonList(
+                  "CREATE TABLE NumericTable (" + "Id NUMERIC NOT NULL" + ") PRIMARY KEY (Id)"));
+      final Database database = op.get(1, TimeUnit.MINUTES);
+      assertThat(database).isNotNull();
+    } finally {
```

Comment:
Could we also add the following to this test:
1. Write a couple of values using mutations.
2. Read these back using a `Read` operation, as those require the user to specify the key(s) to read. That ensures that reading using numeric keys work.
3. Delete some of the values using mutations, as those require the user to specify the key(s) to delete.

Reply:
Added these assertions to the test